app/controllers/pensions/unauthorisedPayments/SurchargeAmountController.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.pensions.unauthorisedPayments
+
+import config.{AppConfig, ErrorHandler}
+import controllers.pensions.routes.PensionsSummaryController
+import controllers.predicates.AuthorisedAction
+import controllers.predicates.TaxYearAction.taxYearAction
+import forms.{AmountForm, FormUtils}
+import models.mongo.PensionsCYAModel
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.PensionSessionService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.{Clock, SessionHelper}
+import views.html.pensions.unauthorisedPayments.SurchargeAmountView
+
+import javax.inject.Inject
+import scala.concurrent.Future
+
+class SurchargeAmountController @Inject()(implicit val mcc: MessagesControllerComponents,
```

Comment:
You could put all of the implicit values together in a separate set. See `OneOffRASPaymentsAmountController`.